core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -2909,6 +2911,37 @@ private File getRootDirFor(String name) {
         return new File(new File(getRootDir(),"jobs"), name);
     }
 
+    public static String urlEncodeMatrixConfigString(String fullName) {
+        String html_escaped_string = "";
+        StringTokenizer parameter_tokens = new StringTokenizer(fullName, "=");
+        while (parameter_tokens.hasMoreTokens()) {
+            String before_equal = parameter_tokens.nextToken();
+            String after_equal = "";
```

Comment:
```suggestion
            String after_equal = null;
```

You are assigning later, no need to initialize it.